The regular test code generates certificates with a 512 bytes long
key, which makes some client libraries unhappy when connecting to
the HTTPS API server.

Index: 2.0.2/src/github.com/juju/juju/testing/cert.go
===================================================================
--- 2.0.2.orig/src/github.com/juju/juju/testing/cert.go
+++ 2.0.2/src/github.com/juju/juju/testing/cert.go
@@ -52,7 +52,7 @@ func verifyCertificates() error {
 }
 
 func mustNewCA() (string, string) {
-	cert.KeyBits = 512
+	cert.KeyBits = 1024
 	caCert, caKey, err := cert.NewCA("juju testing", "1234-ABCD-IS-NOT-A-REAL-UUID", time.Now().AddDate(10, 0, 0))
 	if err != nil {
 		panic(err)
@@ -61,7 +61,7 @@ func mustNewCA() (string, string) {
 }
 
 func mustNewServer() (string, string) {
-	cert.KeyBits = 512
+	cert.KeyBits = 1024
 	var hostnames []string
 	srvCert, srvKey, err := cert.NewServer(CACert, CAKey, time.Now().AddDate(10, 0, 0), hostnames)
 	if err != nil {
